app/models/concerns/course/assessment/submission/answers_concern.rb

```diff
@@ -17,15 +17,38 @@ def from_question(question_id)
   end
 
   def create_new_answers
-    questions_to_attempt ||= assessment.questions
+    # Load questions from submission instead of assessment in case of randomized assessment
+    questions_to_attempt ||= questions.includes(:actable)
```

Comment:
Can I check where this `questions` is coming from?

Reply:
Its defined here:

https://github.com/Coursemology/coursemology2/blob/79f814349ad54ba571533921eff331782eba9fbd/app/models/course/assessment/submission.rb#L185-L187